### ISSUE TITLE
Some ideas for changing the refugee center.

### ISSUE THREAD

**Issue by IdleSol** [CleverRaven/Cataclysm-DDA#71252]
### Is your feature request related to a problem? Please describe.

A little digression. I apologize in advance:
- for the quality of the text, I use Deepl translate;
- for the deviation from the rules;
- for the large amount of text.

There is a problem in the game: unique faction locations are not unique. They can be encountered several times per game. And the longer you explore the world, the more copies you encounter.

In one of the first games, I encountered a second refugee center. At the time, I considered this event a bug. So I started a new world. And met a duplicate again. The game was very frustrating. And led me to suspend the game. Yes, now I just ignore duplicate factions, locations, NPS. That's the compromise I've come to.

I've seen the explanation for why this is done. Because of the possibility of a location being infected with fungaloids. To be sure to encounter the location, regardless of the direction of travel. I don't know how true that is.

But no matter the reason. This situation is still a problem.  That's what my gaming experience tells me.

It was this situation that became the starting point for thinking, "What can be done at all?". As a basis, I took the refugee center. And I want to share the result of my reflections. 

I realize that the developers have their own vision of the situation. So take this as a kind of discussion. Or as a reason to once again remind about the problem.

### Solution you would like.

<details><summary><b>1. Add a number to the name of the refugee center</b></summary>

I suggest renaming the refugee center using the following template: 
- refugee center N<random_number>;
- refugee center <random_number>N;
- refugee center <random_number_1>N<random_number_2>;

Where the number N is an integer from 1 and until you run out of imagination. It will determine the type of refugee center (see point 7) and their features. I immediately suggest to count on N>9. Then N = 01, 02 ... 11 ... 99.

And this point alone, without the realization of the subsequent ones, should improve the situation. Even if only marginally. Since you're not encountering the same refugee center, but a different center. After all, it doesn't have to be the same one, does it? Yes, you will still need to ignore the same NPCs.
</details> 

<details><summary><b>2. Change the appearance of the refugee center (optional)</b></summary>

<b>2.1. You need to add underground floors and move most NPCs to them.</b>

Roughly as follows 

- The surface remains for the lobby: a merchant, a doctor, a couple guards, and the homeless. Maybe a couple of technical rooms.
- The rest of the npc's are moved to level -1, including the death room. 
- Levels -2, -3 are closed to the player, just like level -1 is now. I'm in favor of them being drawn in the game, not missing as they are now. Will be needed (or not) for later items.

<b>Rationale</b>.

Perhaps the main reason is portal storms. At the moment, I don't know of any reasoning for the survival of any faction that doesn't have access to the basement. And that goes for many other factions: Cody, the metalworker, the bandits, the religious community. If a storm hits a location, everyone who didn't hide will die.

So either change the storm, which is unlikely to happen, or change the locations for factions.

Additionally. If the room with the zombies will be on the underground level: steel doors and concrete walls, this might at least somehow explain why they haven't broken out of it yet and killed all the refugees.

<b>2.2. Additional changes:</b> 
- remove shrubs and trees;
- drain water sources;
- the surrounding terrain should not be a swamp, forest or water body (on the global map, surround the center with two tiles of fields).

<b>Rationale</b>.

It is strange to build underground facilities in an area with a lot of water, it causes additional costs during construction and problems during operation. Especially since this is a shelter in case of war. It's easier not to take the risk than to deal with underground flooding later.

The forest, next to the shelter, is potential problems with debris and fire. And for the game, mushroom contamination. Because while in the field you can still kill them, in the forest you can't.
</details> 

<details><summary><b>3. Changing the map generation (optional)</b></summary>

I don't know the mechanism of map generation, so I can't suggest anything specific. We need some filter (?), mask (?), in general some possibility to prevent generation of unique location near "hostile" locations.  By hostile locations I mean: fungaloid, triffid, ant, blobs and open portals that can summon creatures.

<b>Rationale</b>.
If there are enemies near a faction, it doesn't survive and we don't meet it. And if we meet it, then there are no enemies near them.

It's hard for me to somehow estimate what the distance from the center of the faction to the enemies should be. The only thing that comes to mind is that it should be safe, i.e. big enough.

If you look at it from the perspective of game mechanics, it's being able to get to the faction and work near it without the enemies getting caught in the reality bubble.
</details> 

<details><summary><b>4. Duplicating the refugee center in the game files</b></summary>

This is an intermediate point. This is where the number N that I mentioned in point 1 comes in handy.

As a result, we will have N identical "refugee centers N<random_number>"
</details> 

<details><summary><b>5. Adding a restriction on the appearance of duplicates</b></summary>

One more intermediate point. We need the ability to count the number of generated unique locations and the ability to limit the appearance of these locations when some threshold is reached.

Example.

N=1 - this location can appear only 2 times. N=2 - only once, N=3 - no restrictions. And similar variants.
</details> 

<details><summary><b>6. Entering supply and demand for merchants and adding time frames (optional)</b></summary>

Third intermediate point, little relevant to the topic, but necessary to understand the proposed changes.

We need the ability to set priority for merchants (maybe we already have it?). In addition, we need the ability to add a coefficient on merchant prices and the ability to change this priority by timer or quest completion.

<b>Example</b>
A merchant in location N=1 buys only food. The first month of the game, he is willing to buy it at x2. The second month - x1.5, and the third month at the standard price. He is not interested in other categories of items (or x0).  The player completes the quest "food abundance" and then upon completion of the quest, regardless of time, the merchant reduces the purchase price or stops buying the goods. 
For N=2, the merchant is interested in materials and tools.
For N=3, the merchant is only interested in materials made of steel and not interested in plastic.
</details> 

<details><summary><b>7. Customization of refugee centers</b></summary>

Returning to the number N, this number determines the type of refugee center and also serves as a metagame pointer for the player. In fact, the player, if he calculates this pattern, will know in advance what he will encounter in that particular refugee center.

Next, I'll spell out some options I can think of and some additional information for a more complete description.

<details><summary><b>7.1 Refugee Center 1. VIP</b></summary>

Who do you dislike more? The government, heads of corporations, or really, really rich people? Not so important, this center is designed just for those people and their escorts.

This center is fully prepared. It is equipped with advanced production lines, to fulfill every need. Every production line is not that duplicated, but rather has ten times the backup. Their stockpiles are not just plentiful, they are sufficient for tens of years of autonomous existence.

Their supplies aren't just plentiful, they're enough for a decade of autonomous existence.

There are three factions in this version of the location: VIP, guards, and scavenger (~ breadwinner / miners / looters / I mean, an item hunter).

<details><summary><b>Description of fractions</b></summary>
<b>VIPs</b> are those for whom this shelter was intended. They are located in luxurious underground rooms. The player has almost no way to interact with them. It is limited to quest interactions.

They're not stupid. They are well aware of the situation and realize that the Earth is, in fact, doomed. And even in that situation, they have everything they need. Hedonism, that's their new religion. Accordingly, the main interest is entertainment. Alcohol, drugs, delicious food... marloss. That's what they're interested in. Everything else? They've got it all. Survivors aren't interested. Brief description: "bad ass."

VIP <b>guards</b> are those who provide security for the VIP. They have 100% loyalty and although it will decrease over time, it will be decades before any decisive action is taken. This high loyalty is due to the fact that their families live safely inside the shelter. They are fully provided for. And they fulfill the usual roles of maintenance staff (maids, cleaners, mechanics, and so on).

There is some division: elite guards, simple guards and civilian personnel (servants and family members).
Elite guards are more of an internal security service. They are directly engaged in VIP protection and serve as insurance against insubordination of ordinary guards. But again, it's still a long way off. But everything is in your hands, so it's more like: "without player intervention, it's still a long way off".

They're equipped with advanced weapons and armor. Perhaps some experimental weapons for elite guards? You want power armor and laser weapons? Then this is the place to go, but be warned, they don't trade.

The third faction is the <b>scavengers</b>. These are the survivors who were able to make it to the shelter and prove their usefulness. They are outside the perimeter of the center and do not have access to the inside. Their main occupation is scavenging for useful items, in an attempt to prove their usefulness and gain access to a safe haven. Futile endeavors. With very few exceptions, none of them will never get in. All promises, are promises. No VIPs or guards are interested in increasing the population.

On a separate note, no beggars or sick people. The relationship is simple or prove your usefulness or get out. If you don't want to, you'll be kicked out with a kick in the ass. You gonna be pushy or cause trouble? Short conversation: bullet in the head.
</details> 

In game, it should only appear in one instance (if point 5 is implemented).

<details><summary><b>Cosmetic changes</b></summary>

Based on this introduction, the existing center should lapping cosmetic changes. 
The perimeter will be surrounded by a concrete fence, no gaps. There are towers to monitor the perimeter. Barbed wire on the second level of the fence. Maybe automatic turrets and a thicker fence? There is only one road into the center, which is accessed by a checkpoint.
The checkpoint is a structure divided into two parts. Inner part: guards and a vender (or vending machine ~ hub). Outer part: scavengers. The wall and doors are reinforced, the doors leading to the inner part are locked.

On the outside of the property, maybe some distance from the fence, is a tent city of scavengers. Or maybe there isn't. I'm not sure it's necessary. And also that the described faction would allow their location under their walls.

Inner territory. Helicopter pad, some number of hangars (?), garages... in general technical buildings. Most likely all concrete, possibly brick(?). Definitely no wooden buildings. There's a communications tower.

Underground levels, if someone will draw. Luxury apartments and smaller rooms. No barracks. And that's another reason to refuse to accept new residents. No one wants to move from their own rooms to barracks.

Lots of warehouses and technical rooms (by lore, for making everything you need). Separate rooms for generator, including lots of batteries. Lots of checkpoints, guards (turrets?). 

Roof of the center - lots of solar panels. Possibly wind generators.

Player only has access to the outside of the center. Any unauthorized entry into the territory causes enmity with the faction.
</details> 

<details><summary><b>Merchants</b></summary>

The faction has its own electronic (important!) currency: utility points(?), social rating(?). The point is that you can't take them with you and use them elsewhere. They're only used here. And characterize how useful you are to this faction.

Since the main faction is interested in entertainment. Merchants are willing to buy alcohol, drugs and marloss. At the regular price, they realize that many are willing to sell them at a lower price, hoping to save up for a pass. The other categories of goods, are either not bought at all, or at a very low price (x0.1?).

On sale, there are only 3 passes:
- ground level pass (accessible inside the walls, inaccessible to the inner buildings and the main building);
- a pass to the main building ground level;
- pass to the underground levels = residence permit.

Reminder, the faction is not interested in new residents. Prices should be so large as to be rather impossible in principle. If we use the dollar equivalent, something in the form of 100k for the first pass, 1kk for the second and 10kk for the third. It's more of a setback for quests. And will help to exclude situations when a player brings a lot of things and buys a pass.
</details> 

<details><summary><b>Quests</b></summary>

Quest direction: get an item (elite alcohol, drugs), check the location (mansion) and report. Participate in a safari on zombies/people/other faction. And accordingly rescue and/or evacuate such a participant. Establish communication with other centers, perhaps to demand and lead to obedience?

I want to describe the situation with marloss. Surely you have already noticed that I have mentioned it several times. The marloss berry and its derivatives. The pleasure of consumption is matched only by the danger of that consumption.

The hedonist faction is naturally interested in marloss as a source of pleasure. And a rough lineup would look like this.

A player who has marloss with him can mention it when talking to the faction's merchant. What exactly he says depends on the player and what he knows.

<b>Normal player:</b> <i>I found a strange berry, saw an animal eat it and beat in ecstasy, but not sure if it's safe, giant mushrooms and all.</i>
<b>Marloss Gateway:</b> <i>I found a berry that is great fun and practically safe.</i>
<b>Player, with some knowledge (?), possibly a dislike of marloss:</b> <i>I have found an extremely dangerous berry that is addictive and leads to a horrible and painful death.</i>

Regardless of the choice of replica, the merchant is willing to accept a batch of such berries and/or derivatives. But the replica determines the multiplier that will be taken into account when choosing the final result.

When the player delivers the ordered batch, the merchant will offer to volunteer to test the effects of this berry on humans. Or simply, you eat it and we'll see. If you survive, we'll reward you.

Response options are: agree, refuse, and bring a volunteer. The volunteer could be one of the followers or bring a beggar from the original refugee center (there was a mushroom lover there). Any other option.

On refusal, the chain ends. The merchant says he will find another volunteer.

A 1d100 die is rolled. The difficulty is 1*K, where K = depends on the answer about danger. For example: 1 if you say dangerous, 3 if you express caution, and 5 if you say safe. The faction is not dumb, regardless of the answer, it will check. If the check fails, well hello marloss fanatics. Re-mining quests are opened.

If the check is passed, the faction confirms the danger, possibly looking for ways to use it safely. Again mining quests and a way to safely consume, if it fits into the lore at all.

If he agrees, the player is taken to an isolated room, asked to change into the clothes given to him and leave all his own behind the door. After that, the berries are consumed. The player spends 7 days locked up, periodically he is questioned about his well-being (or not, observation with the help of autodoc) and given food. And to keep the player from getting bored, there is a vending machine inside with access to technical literature. In fact, 7 days of reading manuals (new books and new recipes?). From the player's reaction to use, change the odds and roll the dice. Marloss Gateway is able to pass all tests and convince of the safety of use, K=100.
</details> 
</details> 

<details><summary><b>7.2 Refugee Center 2 and 3. Scientist and Technician.</b></summary>

<i>Lore. Originally they were to be one shelter, the main purpose of which was to keep scientific knowledge and subsequent revival of production. Due to some reasons, it was decided to divide it into several parts. 

The "Scientist" class refugee centers were intended for scientists. It has a lot of computer equipment and knowledge for all occasions. They are responsible for preserving knowledge and for high-tech equipment, which is quite difficult to produce from scratch. Accordingly, they had to revitalize into complex chains of production. And with some time spent training survivors from other shelters, light manufacturing.

The Technician class refugee center was for tech workers. It concentrates a lot of production equipment, has a library with specialized literature. Their main task, to restore the most simple chains of production, but still technological. And upon completion of their task, in conjunction with the centers of the class "scientist" recovery and more complex.

Unfortunately, there is a small unaccounted fact in the calculations. To do, it is not enough to know, you must also know how. And those who know how, not always know well the theoretical part. On this circumstance superimposed the chaos of the first days of the cataclysm and a large mortality among the alleged personnel of the centers.</i>

These two refugee centers are similar in many ways. Both of them are almost completely finished. Both centers don't boast large inventories. Really depends on who you compare them to. Their supplies are sufficient for the first time, but not enough for the problem to be postponed.

The leaders of these refuges, from the first day of the cataclysm realize the impending threat, although they do not know the causes of occurrence. Therefore, they take actions to ensure the long-term survival of the refuges.

And one of the first actions, was an attempt to establish communication with the neighboring "brother" shelter. For some unknown reason, it failed. Understaffing forced the management to be more loyal to the outsider survivors. Both shelters happily began accepting new survivors. Although reasonable restrictions were imposed.

<details><summary><b>Description of fractions</b></summary>

The population of these centers is represented by two factions, but this division is very conventional and has little effect on life:
- Scientists or technicians according to the class of the shelter;
- Joiners - survivors who came in the first days of the cataclysm and received shelter.

In addition to them, there are representatives of some other factions and loners in the location. Most of them are people willing to provide their services or items for sale and exchange. Although there are also those who hope to get a place in the center and those who were rejected for one reason or another. Mostly for health reasons and lack of useful skills.

<b>In terms of the game.</b>

<b>Scientists</b> is a center specializing in electronics and applied science. They are willing to sell medical supplies, some chemicals, and electronics. You can also buy/order high tech tools from them. They can make things out of synthetics, ceramics, and rubber. Can find or make protective suits: electricity, radiation. Provide some recipes and access to knowledge (sd cards with textbooks).

And if ever an armor enhancement is implemented, along the lines of adding electricity protection to my favorite cotton socks, that's this way too. I'd prefer to take the recipe though.

Happy to buy food, seeds, and some of the materials needed to support their production. In particular electronics, chemicals, synthetics and rubber, maybe plastic. With a time limit they buy guns, cement, bricks and rebar. Demand for these things closes a month after the cataclysm or with a link to the fulfillment of tasks. Do not buy books, robots, ready-made clothes.

With great joy they will buy books from laboratories, but only in one copy. (Through talking: "I found a book here, not interested?"). After can sell and/or make to order mutagens.

Interested in exody and hub armor. Can provide their own samples after study

Bionic modules are ecstatic and request to bring more for study. Will they make and/or modify them?

Basically, this faction is willing to purchase anything that is not of the ordinary stuff and research it, in the name of science. After which release an analog, maybe worse, but more affordable. Ready to share knowledge, especially with those who helped them acquire that very knowledge.

In general, if you think where a classic mad scientist should live, this is where he belongs.

<b>Technicians</b> is a center specializing in mechanics and fabrication. They are willing to sell some simple tools, work clothes and some materials. You can order/buy steel frames. Since there are machines, they make modern weapons and ammunition (AR-15 or M16 level). Can make survivor level armor or simple armor (scrap metal, steel). Some simple edged weapons (steel). If given recipes, can craft armor and weapons from hardened steel.

Willing to purchase food, medicine, fuel, and most spare parts. Not interested in readymade clothing, armor, weapons, or chemicals. Temporarily interested in books and manuals.

Interested in protective suits. Emphasize protection from electricity and radiation: scientist or hub suits. Once trade supplies from scientists are established, fully stocked.

In general, this place is for people who are good at making things with their hands. Cody and the matalloader would fit right in here.

Scientists and technicians, form a close-knit alliance, coming out to be self-sufficient, food aside. If it weren't for the distance between them and encounters with bandits and other dangers of the world.
</details> 

The number of these escapes in the world is limited. Say no more than 2 pieces of each kind. Feature, you can make a relatively close location and the presence of a road between them with communication towers. That is, they are created at once a pair, of course, if possible.

<details><summary><b>Cosmetic changes</b></summary>

The refugee center is surrounded by a brick fence. In some places there are broken or missing sections that need to be repaired or built. At the entrance, there is a checkpoint that houses guards, a merchant, and some random NPCs.

Before entering the area, a tent city that houses asylum seekers, survivors, and homeless people. There are (or will be) representatives of the traders' union there as well.

Inside the territory is a lot of free space, as you develop (time or quests), will be filled with fields with a transition to greenhouses and other buildings necessary for survival (garage, blacksmith shop and something like that). There are communication towers. 

A roof, a small number of solar panels by scientists. As time passes or the fulfillment of quests, the number will increase. Technicians or by analogy or start without solar, perhaps with homemade wind turbines.

Underground levels are production, shelters, warehouses and generator room. Living quarters are mostly 2 person rooms, there are barracks.

More loyal to strangers than VIPs. To pass through a checkpoint at the entrance. A friendly reputation is required. But underground rooms are available only on quests or are not available at all.
</details> 

<details><summary><b>Quests</b></summary>
Directing missions. Establishing communication with other centers: scientists with technicians, and technicians with scientists. Searching for specialists, materials.

Time-limited tasks. Mending a fence: finding and delivering cement, bricks and rebar. Allocating workers from their followers. 

Restriction on the task until the beginning of summer, June 1 is considered that the quest is completed by the forces of the refugee center. For the scientists, the search for weapons, similarly by early summer, is considered completed by their own forces.

The scientists are willing to study the marloss and give extensive information about its dangers. They are prepared to study a lot of things in general.

Establishing communication with exody, delivery of recipes and spare parts (external climate control kit and CBM interface wire kit) could give an analog powered by UPS or with a built-in battery (as in a smartphone). Ideally, a blueprint for fabrication and the possibility of ordering. Establishing communication with the HUB and analyzing his equipment, could give access to suits with protection from radiation and electricity. And fulfilling both assignments, a climate-controlled protective suit.

In addition, the rescue of valuable specialists, the research of laboratories (coordinates of laboratories) and the need to find and deliver NPCs.

Seed Search. Execution and/or time of season, can be a trigger for the appearance of fields on the refuge. Searching for glass and other materials, for the appearance of greenhouses.
Quests to find and deliver animals: cows, horses, chickens. Possibly repeatable?

To the techs, on the underground level, I would relocate the death room from the original refugee center. And expand it to part of the underground level. Spreading also the number of zombies over a larger area, so that it would be possible to deal with them in the first days.

<i>One of the refugees hid the infection, fearing that he would not be allowed in. In the confusion of the first few days, no one paid attention, and few realized what was happening. During the night, he died and attacked his barracks neighbors. Panic broke out, the corridor was jammed, preventing the guards from getting to the dead man in time.

The result... the zombie was put to rest a second time. Two refugees were bitten. They were treated and isolated in a separate room. And the six refugees who died in the crush were moved to the morgue. We planned to bury them in the morning.

Well, now we know that a zombie bite is optional. In the morning, when relatives and friends came to pick up the bodies, they rose a second time. The tragedy repeated itself on a larger scale. Now a dozen and a half dead are locked in the medical wing, the first underground. Our scavenger squad will be back in about a week and we plan to deal with them. If you want to help, come on down.

What about those bitten in the first incident? Survived and are helping with the work.</i>

Options: refuse, agree to help (group work), and do it all alone. If you agree, the date is ~70 day of spring of 1 year. Along with the player, comes a certain number of NPCs with melee weapons and firearms. If alone, they give you a room pass - to avoid fiddling with lock picks and barricades, it's better to use an electronic pass and lock.

Regardless of the player, on day 71 the quest becomes unavailable or failed and is considered to be completed by the faction's forces.

Analogous to the quest, big spiders.
<i>Hell knows how they got into the manufacturing complex. Maybe they crawled in while they were small, maybe there's a hole somewhere, or maybe they teleported in. The point is, there's a damn giant spider in the underground complex and we need help. Will you do it?</i>
</details> 

</details> 

<details><summary><b>7.3 Refugee Center 4. General Purpose.</b></summary>

This is a general purpose refugee center. It was designed for ordinary people. It's the most common type of shelter. Which is why it was funded on a residual basis.

Unfinished, low on supplies. It's easier to list what's there than to say what's missing. They have beds and some supplies, and they're running out. And they have problems with refugees, with leadership and security.

Basically, it's our classic refugee center. Maybe some of the residents will move in with the techs, along with the death room. Others to the scientists. And maybe split between 41 and 42. Where 41 will be the homeless lobby and 42 will be Alonso's pants.

<details><summary><b>Cosmetic changes</b></summary>
From cosmetic changes, only the gradual replacement of the fence with a wooden version, the appearance of wooden buildings and fields for vegetables on the territory of the shelter. In general, it is strange to have such a territory and not try to grow something on it. Naturally tied to the time and fulfillment of quests. Conditionally one side of the fence for one month without a player or for half of the remaining time, if the player brings boards / logs. Then more boards and more, and more, and many times more.

</details> 

<details><summary><b>Merchants</b></summary>
This center needs everything from food to clothing. Which it would rather recycle to make the items it needs.

Remember the experience of playing with a 0.1 loot factor. Who is not familiar, the algorithm: pick up dirty clothes, wash, dismantle into parts and make new ones from them, is the norm, not the exception. True only at the beginning of the game.

And here an entire shelter is in a similar position. So they buy everything, medicine and food at higher prices because they need it badly. And they sell everything except medicine and food. But the assortment of sales should depend on the time. Certainly in the spring of 1 year, they do not sell anything, in the summer some assortment already begins to appear. In the autumn, they go full steam ahead.
Melee weapons (max steel), bows and rare firearms (AR-15), along with leather armor or some versions of gambesons. Here's the upper bar for the people of center 4.

Maybe they'll be the center of free traders? They can't produce anything worthwhile, so they had to find something to sell and who to sell it to.
</details> 

<b>Optional.</b> I suggest removing the Tacoma commune from the game. Move quests and buildings to territory 4. Cut down the forest and plow the fields around the refugee center. Move some buildings inside the perimeter. Add barracks with beds. For the refugees, it's a food source, for the commune, it's shelter during a storm.
In the event of a cataclysm, I don't see the point in moving to the commune. When you can stay in a relatively safe place and plow the fields nearby. Especially with portal storms like this.

</details> 
<details><summary><b>7.4 Refugee Center 5..7. Mushrooms.</b></summary>

Here's a center that's not so lucky. We take 1, 2, 4 and infect with fungaloids. The result is center 5..7. Number 3 is omitted, as it is similar to number 2 and after mushroom infection, one can hardly find differences. Accordingly there must be a mushroom tower somewhere nearby.

These versions of the centers are for mid to late game. Exploration and looting. So it is worth thinking about some reward.

For the mushroom version of the VIP hideout (number 1), it could be a preserved room with valuable loot. Or the location itself could be a reward for turning the inhabitants into marloss fanatics. They became fanatics, infecting everything with mushrooms. And when the player returns there, instead of the usual, he sees this option.

For scientists (number 2) - a preserved terminal with access to the digital library and rare blueprints, if unique blueprints are implemented for them.

For general purpose centers (number 4) - you can add nothing. They don't have anything of value. Just devastation and a lot of fungaloids.

If you want, you can house survivors in them, with a timer until death. Something like a couple months. They started the launch by challenging the lab. They managed to haul in a couple months' worth of supplies. And then the fungus arrived and there was nothing left for them to do but hope for help.
</details> 

<details><summary><b>7.5. Refugee Center 8..10. Zombies</b></summary>

It's zombie version 2, 3, 4. I doubt center number 1 can be turned into zombies, too many heavily armed guards there. 

Zombie version 3 (Technicians) or 4, this is a variant of the failed death room cleanup. The zombies broke out of the room, the cleanup failed... whatever, now it's just the dead.
</details> 

<details><summary><b>7.6 Refugee Center 11. Other</b></summary>

Any option that comes to your mind.
</details> 
</details> 

<details><summary><b>8. The relationship between NPCs and hostile location generation</b></summary>

In point 3, I suggested limiting the possibility of a hostile location appearing near a unique location.

But let's imagine the situation: "You are in charge of a refugee center. And one not-so-good day, you see a large tower of Fungaloids on the horizon." What are your actions?

Are you strong and have lots of resources? Send a team to destroy this threat.
You're not sure you have enough strength? Ask passing survivors to join your team or perform instead of you.
You are weak... you have died or escaped.

Ideally, if there is a threat near the npc camp, they should react to it. Destroy it, issue quests for it, or run from it.

How can destruction be implemented? At least we need a sequence of generation: first a world with NPC locations is generated, and then a threat is generated in this world. At a certain distance, the threat is replaced by a variant of the destroyed threat. That is, instead of a mushroom tower, a large ashes are generated, and instead of fungoloids their corpses. And then the player walking by will see exactly that, someone has destroyed the threat. In a conversation NPC, can tell that yes, it was interfering with us and we destroyed it.

Similarly, you can realize the possibility of issuing a quest. If the distance from the camp to the threat is less, then the option to issue a quest becomes available. Or several quests.

Have you escaped? Instead of a base camp - ruins. Somewhere nearby, you can find a bunch of refugees who were in that camp. With brand new quests. Or a quest: destroy the threat and clear our base.
</details>

**Comment by Rabadash94** [comment]
The Refugee Center is already globally unique in the experimental version.

**Comment by db48x** [comment]
As already mentioned, this has already been fixed in the experimental builds and in the 0.H version that is coming out soon.

There is also talk of long–term work towards a system of world generation that puts certain NPC faction locations closer to or further away from certain types of threats, as well as each other. See #70056 for some discussion of the possibilities; you will probably be really interested in them.

**Comment by kevingranade** [comment]
As noted already,  it's already been marked as globally unique,  which is the core issue here.

Regarding the multiple other suggestions you've made,  issues must address one issue,  it's impossible to track a bunch of topics in one issue,  it just leads to issues that stick around forever if any of the topics are not addressed, and they frequently even start accumulating more topics via discussion.

In broad strokes,  a lot of your suggestiond boil down to "make multiple refuge centers",  which is actually fine,  but we don't need ideas for what that would look like,  we need people that are willing to put in the work to actually add that content.  It's also not at all clear why we need some special system for variations of refuge centers,  just start with the existing building,  make a copy,  then start editing it to match your plan. There are techniques to reduce the amount of content you have to produce to make this happen by sharing palettes,  nested mapgen chunks,  item groups, etc, but any raccoons "alternate refuge center" is going to have so much different about it that you're going to need to rebuild it from the ground up.
